### DISCUSSION TITLE
Hide chart tab for users without permission to edit charts

### DISCUSSION THREAD

**Discussion by fmannhardt** [apache/superset#23802] [Ideas]
I am a bit unsure whether to log this as a feature request or a bug. When a user without the permission to edit/save charts logs in they are still presented with the list of charts to explore. However, when following any of the chart links, the user can obviously not do anything useful or (depending on the permissions) not even interact with the view.

![afbeelding](https://user-images.githubusercontent.com/3202998/234140634-83e9bb88-294d-4c67-bb06-3aa2bf220b1b.png)

I could create a simple PR (already prepared) that hides the charts on the welcome page as it is already done for the saved queries. I based it on the `can_write` permission for charts but I am a bit unsure if that is the only one that this behavior should apply for.

**Comment by mickymrt** [comment]
Hi
I have the same issue, did you find a workaround?

**Comment by Seddawy1992** [comment]
did anyone find a solution for it to hide them from the welcome page?

**Comment by martimors** [comment]
This would be nice, currently users with no chart creation are presented with charts on the welcome page but if they click on them they get a 403 error. Quite confusing for the bulk of our users who are just dashboard explorers.

I suggest reusing `menu_access on Charts`, if user lacks that one then charts should be hidden. Maybe the same for Dashboards too? @sfirke would you accept a PR on this?

**Reply by sfirke** [reply]
@martimors absolutely! That would be great, feel free to ping me on it as it progresses.

I like the idea of tackling the dashboard case too but I'm trying to think of a scenario where someone has Superset access but not access to dashboards or charts. Is there one?

**Reply by martimors** [reply]
Checking back to say that I am still planning to submit a fix for this at some point. 🙏

**Reply by martimors** [reply]
@sfirke This turned out to be potentially much easier than I had anticipated, here's my changeset 😎  https://github.com/apache/superset/pull/33946

**Comment by LockedInTheSkage** [comment]
Hey @martimors and @sfirke, did you find a solution to this problem?

**Reply by sfirke** [reply]
It looks like we (Superset committers) neglected @martimors pull request above (sorry!) but now it is getting attention again.

**Reply by LockedInTheSkage** [reply]
Great! Looking forward to following this issue:)
